src/main/java/com/cloudbees/hudson/plugins/folder/computed/PeriodicFolderTrigger.java

```diff
@@ -43,12 +46,21 @@
  */
 public class PeriodicFolderTrigger extends Trigger<ComputedFolder<?>> {
 
+    private static final Logger LOGGER = Logger.getLogger(PeriodicFolderTrigger.class.getName());
+
     /**
      * The interval between successive indexings.
      */
     private final long interval;
 
     /**
+     * Timestamp when we last {@link #run}.
+     * Normally we rely on {@link FolderComputation#getTimestamp} but there is a slight delay
+     * between {@link ComputedFolder#scheduleBuild(int, Cause)} and {@link ComputedFolder#createExecutable}.
+     */
+    private transient long lastTriggered;
```

Comment:
Should note that this does not need to be volatile as the contract from `FolderCron` is that only one call to `run()` will be in flight at any point in time